src/actions/channels.js

```diff
@@ -662,6 +662,18 @@ export function deleteChannel(channelId) {
 
 export function viewChannel(channelId, prevChannelId = '') {
     return async (dispatch, getState) => {
+        const {currentUserId} = getState().entities.users;
+
+        const {myPreferences} = getState().entities.preferences;
+        const viewTimePref = myPreferences[`${Preferences.CATEGORY_CHANNEL_APPROXIMATE_VIEW_TIME}--${channelId}`];
+        const viewTime = viewTimePref ? parseInt(viewTimePref.value, 10) : 0;
+        if (viewTime < new Date().getTime() - (3 * 60 * 60 * 1000)) {
+            const preferences = [
+                {user_id: currentUserId, category: Preferences.CATEGORY_CHANNEL_APPROXIMATE_VIEW_TIME, name: channelId, value: new Date().getTime().toString()}
+            ];
+            savePreferences(currentUserId, preferences)(dispatch, getState);
+        }
```

Comment:
Unfortunately the `last_viewed_at` time that the server keeps track of doesn't work. That time is actually the time of the last post in the channel when you view it. So I'm storing another preference here with an approximate view time. (Approximate because there's no need to add an additional request to every channel view.)

Reply:
I'm not particularly happy with so I'm open to alternatives. I think ideally the members table would probably have another column with the actual wall time, and the /view API would keep those up-to-date.

Reply:
Hmmm yeah, I think this is fine for now but you're right ideally this is handled in the backend by a change to the members table